fe/fe-core/src/main/java/org/apache/doris/nereids/StatementContext.java

```diff
@@ -149,6 +152,10 @@ public class StatementContext implements Closeable {
     // Record table id mapping, the key is the hash code of union catalogId, databaseId, tableId
     // the value is the auto-increment id in the cascades context
     private final Map<TableIdentifier, TableId> tableIdMapping = new LinkedHashMap<>();
+    // Record the materialization statistics by id which is used for cost estimation.
+    // Maybe return null, which means the id according statistics should calc normally rather than getting
+    // form this map
+    private final Map<RelationId, Statistics> relationIdToStatisticsMap = new LinkedHashMap<>();
```

Comment:
how to deal with rollup? the selectedIndex is not part of key

Reply:
i think u just need to replace statistic derive result of mv olap scan by this value if it present?

Reply:
selectedIndex  is a new scan, relationId is different from original scan, should be also getted from the `relationIdToStatisticsMap`